src/rewrite.jl

```diff
@@ -79,6 +79,12 @@ function Base.:/(z::Zero, x::Any)
     end
 end
 
+Base.iszero(::Zero) = true
+Base.isone(::Zero) = false
+Base.:(==)(::Zero, x) = iszero(x)
+Base.:(==)(x, ::Zero) = iszero(x)
```

Comment:
See the ambiguities in the `Aqua` CI job.

Methods like this are unfortunately quite hard to write because you just run into a bunch of ambiguities. Julia lacks the concept of an ordering of the method definitions, so you can't say "No, I reeeeeeaallly want this method prioritized over all others".

As another issue, this method assumes that `iszero(x)` is defined.

So `"a" == MA.Zero()` will fail with 
```
ERROR: MethodError: no method matching zero(::String)
```
which violates JuMP's MethodError principle: https://jump.dev/JuMP.jl/dev/developers/style/#Dealing-with-MethodErrors

I'm okay to keep these methods:
```julia
Base.iszero(::Zero) = true
Base.isone(::Zero) = false
```
but they need tests.

Reply:
Ok, I removed the comparisons - though I'm not sure if this is better than trying to fix the ambiguities with Base that Aqua reports. Of course, one or the other package might at some point produce further collisions which have to be taken into account, but this is more or less unavoidable due to Julia's method resolution.
I agree that the method error for string is confusing. But do you want to apply JuMP's policy to MutableArithmetics? I would say that if a novice wants to deal with the whole concept of mutable arithmetic types, he's probably no novice any more ;). Just a couple of thoughts.